### DISCUSSION TITLE
Grove - High Temperature Sensor | Seeed Studio Wiki

### DISCUSSION THREAD

**Discussion by giscus** [Seeed-Studio/wiki-documents#2036] [Q&A]
# Grove - High Temperature Sensor | Seeed Studio Wiki

Grove - High Temperature Sensor

https://wiki.seeedstudio.com/Grove-High_Temperature_Sensor/

**Comment by AndoRandria** [comment]
Hi,

I bought one and I am trying to read the temperature on my ESP32-S2-DevKitC1 but the value indicate "nan".
I am using Arduino IDE 2.3.4 and copy the code provided here for Arduino but changed the PIN according to my wiring. All wiring check done, but I don't understand why the value indicate "nan". 

Need some help, thanks

**Reply by nfs0619** [reply]
Thanks for reaching out! The "nan" value typically indicates an issue with the sensor reading, possibly due to incorrect wiring or a pin mismatch. We recommend checking the following:

1. Ensure that the correct pin is defined in the code, and that the wiring matches the pin configuration.
2. Double-check the sensor's power supply and ground connections.

If the issue persists, feel free to contact our technical support team at techsupport@seeed.io. They will be happy to assist you further with troubleshooting.

**Reply by AndoRandria** [reply]
Thank you nfs0619 for your feedback.

I have ensured the correct pin is defined in the code, and also double-check the sensor's power supply and ground connections.
But the issue still persists. I contacted the technical support team and will see what they propose. I will keep you informed

Thanks
